s3_to_logsene.py

```diff
@@ -19,37 +23,38 @@
 
 def send_to_logsene(text, logsene_connection):
   to_send = "aws-lambda " + LOGSENE_APP_TOKEN + ":" + text + "\n"
-  logsene_connection.send(to_send)
-  #print "Sent string: " + to_send
+  logsene_connection.send(to_send.encode())
+  logger.debug("Sent string: " + to_send)
   
 
 def lambda_handler(event, context):
     logsene_connection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     logsene_connection.connect((LOGSENE_SERVER, LOGSENE_PORT))
     
-    print "Received event: " + json.dumps(event)
+    logger.debug("Received event: " + json.dumps(event))
 
     # Get the object from the event and show its content type
     bucket = event['Records'][0]['s3']['bucket']['name']
-    key = urllib.unquote_plus(event['Records'][0]['s3']['object']['key']).decode('utf8')
+    key = urllib.parse.unquote_plus(event['Records'][0]['s3']['object']['key'])
     try:
         response = s3.get_object(Bucket=bucket, Key=key)
         body = response['Body']
         data = body.read()
     except Exception as e:
-        print 'Error getting object {} from bucket {}. Make sure they exist and your bucket is in the same region as this function.'.format(key, bucket)
+        logger.error(f'Error getting object {key} from bucket {bucket}. Make sure they exist and your bucket is in the same region as this function.')
```

Comment:
I assume that 'f' there is correct?

Reply:
Yes, it's string formatting. The syntax changed in Python 3, now it's like:

```
>>> a=7
>>> print(f'Number is {a}, yo!')
Number is 7, yo!
```